rust/kernel/of.rs

```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! Devicetree and Open Firmware abstractions.
+//!
+//! C header: [`include/linux/of_*.h`](../../../../include/linux/of_*.h)
+
+use alloc::boxed::Box;
+
+use crate::{
+    bindings, c_types,
+    error::{Error, Result},
+    types::PointerWrapper,
+    CStr,
+};
+
+use core::mem::transmute;
+
+/// Wraps a kernel Open Firmware / devicetree match table.
+///
+/// Rust drivers may create this structure to match against devices
+/// described in the devicetree.
+///
+/// # Invariants
+///
+/// The last array element is always filled with zeros (the default).
+pub struct OfMatchTable(Box<[bindings::of_device_id; 2]>);
```

Comment:
For public APIs, we're trying to avoid dependencies on `bindings`. In fact, eventually we want to make `bindings` private.

So I suggest a newtype here that wraps `bindings::of_device_id`. Or even some Rust struct that is easy to use by clients, accompanied by a (private) `const` function that converts that into an array of `bindings::of_device_id` -- which is what we do with file operations, for example.

Reply:
Hi @wedsonaf thank you so much for the review, really appreciate it !

I'm not sure I understand your feedback - the `bindings::` member inside the public `OfMatchTable` is private, so clients outside `of.rs` don't know, see or care that it contains a (private) `bindings::` type?

My apologies if I have misunderstood. Could you provide an example of what your concern is? That usually helps me "see the light".

Reply:
Oh, my bad. For some reason I read this as a type alias instead of a new type. I even suggested that you do exactly what you're already doing.